frontend/components/Message.tsx

```diff
@@ -1,9 +1,57 @@
 import { getNickCSSClass } from '../data/UserHash';
+import { hasTrailingChar, splitToTextWithoutTrailingAndTrailing } from '../util/stringUtils';
 
 export default function Message(props: IMessage): JSX.Element {
 
   /**
-   * Returns a message formatted in the way irssi formats it for /me
+   * Wraps an a href around text, given an url and clickable text.
+   */
+  function formatLinkNormally(href: string, clickable: string): JSX.Element {
+    return <span> <a href={href}>{clickable}</a></span>;
+  }
+
+  /**
+   * 
+   * Wraps an a href around text, given an url and clickable text,
+   * but excludes a trailing , or . from the clickable link.
+   */
+  function formatLinkWithEndChar(href: string, clickable: string, endChar: string): JSX.Element {
+    return <span> <a href={href}>{clickable}</a>{endChar}</span>;
+  }
+
+  /**
+   * Calls either link formatting method depending on whether the href or clickable text contains trailing characters.
+   * Also splits up the href and clickable into parts with and without the trailing characters.
+   */
+  function formatLink(href: string, clickable: string): JSX.Element {
+    if (hasTrailingChar(clickable) || hasTrailingChar(href)) {
+      const [clickableWithoutEndChar, endChar] = splitToTextWithoutTrailingAndTrailing(clickable);
+      const hrefWithoutEndChar = splitToTextWithoutTrailingAndTrailing(href)[0];
+      return formatLinkWithEndChar(hrefWithoutEndChar, clickableWithoutEndChar, endChar);
+    }
+    return formatLinkNormally(href, clickable);
+  }
+
+  /** 
+   * If a message contains URLs or (/)r/<subreddit> those parts will be linked.
+   */
+  function linkifyMessageContent(message: string): JSX.Element[] {
+    return message.split(' ')
+      .map(messagePart => {
+        if (messagePart.startsWith('r/')) {
+          return formatLink('https://reddit.com/' + messagePart, messagePart);
+        } else if (messagePart.startsWith('/r/')) {
+          return formatLink('https://reddit.com' + messagePart, messagePart);
+        } else if (messagePart.startsWith('https://')) {
```

Comment:
you should match on the general scheme `://` to link everything and not just https. Have a look at the URI rfc https://tools.ietf.org/html/rfc3986#page-7

Reply:
I deliberately only chose to use `https://` so we don't link to insecure `http` URLs, also didn't wanna bother with prepending the protocol to `www` or whatever. I also deemed linking email addresses and phone numbers out of scope and questionably useful at best.

Reply:
leaving out the mailto: etc urls is fine to me, they're not so clear to parse anyways, but I'd at least support the `<schema>:// ` ones in a general manner, caring about the protocol they are is not our job imo